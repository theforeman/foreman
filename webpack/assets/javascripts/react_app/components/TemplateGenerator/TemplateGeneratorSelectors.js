const selectTemplates = state => state.templates;

const selectGeneratingPropsFromTemplates = ({
  polling,
  dataUrl,
  generatingError,
  generatingErrorMessages,
}) => ({
  polling,
  dataUrl,
  generatingError,
  generatingErrorMessages,
});

export const selectGeneratingProps = state =>
  selectGeneratingPropsFromTemplates(selectTemplates(state));
