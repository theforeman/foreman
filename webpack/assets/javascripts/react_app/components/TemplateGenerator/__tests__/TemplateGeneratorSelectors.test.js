import { selectGeneratingProps } from '../TemplateGeneratorSelectors';

const state = {
  templates: {
    polling: false,
    dataUrl: '/some/url',
    generatingError: 'there was an error',
    generatingErrorMessages: [{ error: { message: 'aha' } }],
  },
};

describe('TemplateGeneratorSelectors', () => {
  it('selects template properties', () => {
    expect(selectGeneratingProps(state)).toMatchSnapshot();
  });
});
