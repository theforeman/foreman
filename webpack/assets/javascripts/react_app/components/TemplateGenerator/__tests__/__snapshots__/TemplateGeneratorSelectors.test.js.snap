// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateGeneratorSelectors selects template properties 1`] = `
Object {
  "dataUrl": "/some/url",
  "generatingError": "there was an error",
  "generatingErrorMessages": Array [
    Object {
      "error": Object {
        "message": "aha",
      },
    },
  ],
  "polling": false,
}
`;
